src/java.base/share/classes/java/io/RandomAccessFile.java

```diff
@@ -1030,17 +1030,36 @@ public final double readDouble() throws IOException {
      */
 
     public final String readLine() throws IOException {
+        if (jfrTracing && FileReadEvent.enabled()) {
+            return traceImplReadLine();
+        }
+        return implReadLine();
+    }
+
+    private String traceImplReadLine() throws IOException {
```

Comment:
I think it would be cleaner to have the FileReadEvent usage in readLine. That would reduce this to readLine + implReadLine, no traceImplReadLine.

Reply:
Refactored the code.